scripts/py_matter_idl/matter_idl/matter_idl_types.py

```diff
@@ -294,6 +301,7 @@ class Endpoint:
 class Idl:
     clusters: List[Cluster] = field(default_factory=list)
     endpoints: List[Endpoint] = field(default_factory=list)
+    device_types: List[DeviceType] = field(default_factory=list)
```

Comment:
We generally use this as a parse backend for `.matter` idl files. 

Zap contains the same info and some more, however the storage is specifically for .matter and I would like to not diverge and add zap-specific items only. If you do want to add this, please update matter idl format and parser to express the information you want to capture.

I am unsure if device types (name, type and id) is something that should be inside a matter file.